### PR TITLE
First pass at addng GENIE info

### PR DESCRIPTION
**Pull request by krwood**
Added the GENIE header/summary information to the ndflow output

References between GENIE interactions and tracks/segments and the trajectories seems to have worked.

Applied same unit and axes conversions to the GENIE interaction vertex to match the trajectories.

Modified how the truth info slices were calculated to be uniform, fix at least one bug, and to be simpler (in my opinion). The original methods are left in as commented code.

### CODE REVIEW

File: src/proto_nd_flow/reco/charge/raw_event_generator.py [hunk 1]
```diff
@@ -154,17 +154,25 @@ def _convert_mc_truth_tracks(self, tracks):
...
-            traj_copy[field][:, 1] = traj[field][:, 1] * units.mm + 218.236
-            traj_copy[field][:, 2] = traj[field][:, 0] * units.mm
+            traj_copy[field][:, 0] = traj[field][:, 2] * units.cm
+            traj_copy[field][:, 1] = traj[field][:, 1] * units.cm + 21.8236
+            traj_copy[field][:, 2] = traj[field][:, 0] * units.cm
         for field in ('pxyz_start', 'pxyz_end'):
-            traj_copy[field][:, 0] = traj[field][:, 2] * units.MeV
-            traj_copy[field][:, 1] = traj[field][:, 1] * units.MeV
-            traj_copy[field][:, 2] = traj[field][:, 0] * units.MeV
+            traj_copy[field][:, 0] = traj[field][:, 2] * units.keV
```
Comment (krwood) [inline src/proto_nd_flow/reco/charge/raw_event_generator.py]: What is the rationale for keV instead of MeV here?

Comment (cuddandr) [reply src/proto_nd_flow/reco/charge/raw_event_generator.py]: Running ndflow with the units set to MeV for the trajectories gives output that is 1000x too large for the momentum (e.g. pxyz_start). The attached plot shows the muon initial momentum distribution and note the axis scale, the peak is at ~5E6 MeV or 5 TeV. This is present in the latest batch of MiniRun3 Flow files and from running ndflow myself over the MiniRun3 larndsim files.

Changing the units to keV fixes this, however looking at the larndim files the trajectory momentum seems to be correctly in MeV... so I'm not sure if there is some other conversion being applied to the units.

image

Comment (cuddandr) [reply src/proto_nd_flow/reco/charge/raw_event_generator.py]: I will also note here (so as not to forget) that I need to double check the y-offset and the position units for trajectories, and if the GENIE vertex needs the y-offset applied.

**Comment by krwood** [comment]
@mjkramer @cuddandr Should we merge this into MiniRun3_dev?

**Comment by krwood** [comment]
Nevermind, I see this was already done https://github.com/DUNE/ndlar_flow/commit/cb8594401eaa43cbff68cb69634a20f3a66c56ec

I'm just going to delete this branch since we have it in MiniRun3 (which we should merge into develop at some point)

**Pull request closed without merging**